### PR TITLE
[BACK-86] Change is_bookmarked state upon addition or deletion of bookmarks

### PR DESCRIPTION
**Pull request by thomas-woehrle**
### Description of the issue
Currently the resolvers responsible for managing the bookmarks only change the bookmark table. 
However they should also change the is_bookmarked state of repos upon insertion or deletion across all users. 
Otherwise bookmarked repositories won't be updated

### How I implemented
- create method that updates the `is_bookmarked` column of a project, by looking at the bookmark table
- let the resolvers for adding and deleting the bookmarks call that function 

### Other 
none

### CODE REVIEW

File: packages/graphql_server/src/graphql/resolver/bookmark.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+} from '../../supabaseUtils'
+
+/**
+ * Bookmarks a projects for a user.
+ * @param {string} userID - The user ID of the user in question.
+ * @param {string} projectID - The project ID of the project in question.
+ * @param {string} category - The category the bookmark should be added to.
+ */
+export const addBookmark = async (userID: string, projectID: string, category: string) => {
+  if (await bookmarkIsAlreadyInDB(userID, projectID)) {
```
Comment (pczern) [inline packages/graphql_server/src/graphql/resolver/bookmark.ts:14]: why is this check needed when we could use upsert to update and insert?

File: packages/graphql_server/src/graphql/resolver/bookmark.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    return {
+      message: 'This bookmark is already in the database.',
+      code: '409'
+    }
+  }
+
+  const insertionError = await insertBookmark(projectID, userID, category)
```
Comment (pczern) [inline packages/graphql_server/src/graphql/resolver/bookmark.ts:21]: why does insertBookmark method not updatebookmark state?

File: packages/graphql_server/src/graphql/resolvers.ts [hunk 3]
```diff
@@ -17,16 +23,35 @@ const resolvers = {
...
   Mutation: {
     // takes in variables. Parent object _ is never used
-    addProjectByName: async (_: unknown, { name, owner }: { name: string; owner: string }) => {
-      return await createProject(name, owner, null)
+    addProjectByName: async (
+      _: unknown,
+      { name, owner, bookmarkCategory }: { name: string; owner: string; bookmarkCategory: string },
+      context: MercuriusContext
+    ) => {
+      //Todo: refactor user check
```
Comment (pczern) [inline packages/graphql_server/src/graphql/resolvers.ts:31]: @TODO

File: packages/graphql_server/src/graphql/resolvers.ts [hunk 4]
```diff
@@ -68,6 +83,7 @@ const resolvers = {
       }
 
       const deletionError = await deleteBookmark(userID, projectID)
```
Comment (pczern) [inline packages/graphql_server/src/graphql/resolvers.ts:85]: why is the mehod not in deleteBookmark?

**Review by acreturs — APPROVED** [review]

**Review by 2015builder — APPROVED** [review]

**Pull request merged**